src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -414,6 +415,10 @@ typedef struct WorkerPool
 	 * use it anymore.
 	 */
 	WorkerPoolFailureState failureState;
+
+	/* execution statistics per pool, in msecs */
+	double totalTaskExecutionTime;
```

Comment:
Maybe make this a `uint64` too, then you don't get weird float precision problems.